src/hooks/auditoriumSections.tsx

```diff
@@ -0,0 +1,273 @@
+
+import { useSelector } from "./useSelector";
+import { useFirestoreConnect, isLoaded } from "./useFirestoreConnect";
+import { useRecentVenueUsers } from "./users";
+import { useUser } from "./useUser";
+import { GetUserByPostion, useGetUserByPosition } from "./useGetUserByPosition";
+import { useShowHide } from "./useShowHide";
+
+const useConnectAuditoriumSections = (venueId?: string) => {
+  useFirestoreConnect(() => {
+    if (!venueId) return [];
+
+    return [
+      {
+        collection: "venues",
+        doc: venueId,
+        subcollections: [{ collection: "sections" }],
+        storeAs: "currentAuditoriumSections",
+      },
+    ];
+  });
+};
+
+export interface UseAuditoriumSectionProps {
+  venueRowsCount?: number;
+  venueColumnsCount?: number;
+  sectionId?: string;
+  venueId?: string;
+}
+
+export const useAuditoriumSection = ({
+  venueId,
+  sectionId,
+  venueRowsCount,
+  venueColumnsCount,
+}: UseAuditoriumSectionProps) => {
+  useConnectAuditoriumSections(venueId);
+
+  const { userWithId } = useUser();
+  const userId = userWithId?.id;
+
+  const { recentVenueUsers } = useRecentVenueUsers();
+
+  const sectionsById = useSelector(currentAuditoriumSectionsByIdSelector);
+  const section = sectionId ? sectionsById?.[sectionId] : undefined;
+
+  const baseRowsCount =
+    section?.rowsCount ?? venueRowsCount ?? SECTION_DEFAULT_ROWS_COUNT;
+  const baseColumnsCount =
+    section?.columnsCount ?? venueColumnsCount ?? SECTION_DEFAULT_COLUMNS_COUNT;
+
+  // Video takes 1/3 of the seats
+  const videoWidthInSeats = Math.max(
+    Math.floor(baseColumnsCount / 3),
+    SECTION_VIDEO_MIN_WIDTH_IN_SEATS
+  );
+
+  // Keep the 16:9 ratio
+  const videoHeightInSeats = Math.ceil(videoWidthInSeats * (9 / 16));
+
+  const seatedUsers = getAuditoriumSeatedUsers({
+    auditoriumUsers: recentVenueUsers,
+    venueId,
+    sectionId,
+  });
+
+  const getUserBySeat = useGetUserByPosition({
+    venueId,
+    positionedUsers: seatedUsers,
+  });
+
+  const takeSeat: (
+    gridPosition: GridPosition
+  ) => Promise<void> | undefined = useCallback(
+    ({ row, column }: GridPosition) => {
+      if (!sectionId || !venueId || !userId) return;
+
+      return setGridData({
+        venueId,
+        userId,
+        gridData: { sectionId, row, column },
+      });
+    },
+    [sectionId, venueId, userId]
+  );
+
+  const leaveSeat: () => Promise<void> | undefined = useCallback(() => {
+    if (!venueId || !userId) return;
+
+    return setGridData({ venueId, userId, gridData: undefined });
+  }, [venueId, userId]);
+
+  const checkIfSeat = useCallback(
+    ({ row, column }: GridPosition) => {
+      const covertedRowCoordinate = convertToCartesianCoordinate({
+        index: row,
+        totalAmount: baseRowsCount,
+      });
+      const convertedColumnCoordinate = convertToCartesianCoordinate({
+        index: column,
+        totalAmount: baseColumnsCount,
+      });
+
```

Comment:
@sunny-viktoryia it was discussed in the previous comments and the answer was no, that feature is not something we want to include now, but it can be useful to have it in the future though.